web/src/beta/features/Editor/tabs/story/LeftPanel/ContentPage/PageItem.tsx

```diff
@@ -0,0 +1,149 @@
+  onPageDelete?: () => void;
+  onPropertyUpdate?: (
+    propertyId?: string,
+    schemaItemId?: string,
+    fieldId?: string,
+    itemId?: string,
+    vt?: ValueType,
+    v?: ValueTypes[ValueType],
+  ) => Promise<void>;
+};
+
+const PageItem = ({
+  isSelected,
+  isOpenAction,
+  title,
+  hasEmptySpace,
+  propertyId,
+  storyPage,
+  onItemClick,
+  onActionClick,
+  onOpenChange,
+  setOpenedPageId,
+  onPageDelete,
+  onPropertyUpdate,
+}: PageItemProps) => {
+  const t = useT();
+  const [isEditing, setIsEditing] = useState(false);
+  const [newValue, setNewValue] = useState(title);
+
+  const [handleSingleClick, handleDoubleClick] = useDoubleClick(
+    () => onItemClick?.(),
+    () => setIsEditing(true),
+  );
+
+  const handleChange = useCallback((newTitle: string) => setNewValue(newTitle), []);
+
+  const handleTitleSubmit = useCallback(() => {
+    setIsEditing(false);
+    if (newValue?.trim() !== "") {
+      const schemaGroupId = storyPage.property.items?.[1]?.schemaGroup;
+      onPropertyUpdate?.(propertyId, schemaGroupId, "title", undefined, "string", newValue);
+    }
+  }, [newValue, onPropertyUpdate, propertyId, storyPage.property.items]);
+
+  const handleEditExit = useCallback(
+    (e?: React.KeyboardEvent<HTMLInputElement>) => {
+      if (title !== newValue && e?.key !== "Escape") {
+        handleTitleSubmit();
+      } else {
+        setNewValue(title);
+      }
+      setIsEditing(false);
+    },
+    [title, newValue, handleTitleSubmit],
+  );
+
+  return (
+    <ListItem
+      border
+      actionPlacement="bottom-start"
+      isSelected={isSelected}
+      isOpenAction={isOpenAction}
+      onItemClick={handleSingleClick}
+      onActionClick={onActionClick}
+      onOpenChange={onOpenChange}
+      actionContent={
+        <PopoverMenuContent
+          size="sm"
+          items={[
+            // {
+            //   icon: "copy",
+            //   name: "Duplicate",
+            //   onClick: () => {
+            //     setOpenedPageId(undefined);
+            //     onPageDuplicate(storyPage.id);
+            //   },
+            // },
+            {
+              icon: "trash",
+              name: "Delete",
+              onClick: () => {
+                setOpenedPageId?.(undefined);
+                onPageDelete?.();
+              },
+            },
+          ]}
+        />
+      }>
+      {isEditing ? (
+        <StyledTextInput
+          value={newValue}
+          timeout={0}
+          autoFocus
+          onChange={handleChange}
+          onExit={handleEditExit}
+          onBlur={handleEditExit}
+        />
+      ) : (
+        <PageTitle onDoubleClick={handleDoubleClick}>
+          {hasEmptySpace || !title ? t("Untitled") : title}
+        </PageTitle>
+      )}
+    </ListItem>
+  );
+};
+
+const StyledTextInput = styled(TextInput)`
+  width: 100%;
+  font-size: 12px;
+  color: ${({ theme }) => theme.content.main};
+  font-style: normal;
+  font-weight: 400;
+  line-height: 20px;
```

Comment:
Same here, I don't htink we shold have different font/line settings for each TextInput. If the styles of TextInput don't match Figma, updating the input directly is preferable